assets/js/blocks/products/base-utils.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { getRegisteredInnerBlocks } from '@woocommerce/blocks-registry';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ProductListTitle,
+	ProductListPrice,
+	ProductListButton,
+	ProductListImage,
+	ProductListRating,
+	ProductListSummary,
+	ProductListSaleBadge,
+} from '../../atomic/components/product-list';
+
+// @todo how to support these using mapping?
+//'core/paragraph',
+//'core/heading',
+/**
+ * Map blocks names to components.
+ *
+ * @param {string} blockName Name of the parent block. Used to get extension children.
+ */
+export const getBlockMap = ( blockName ) => ( {
+	'woocommerce/product-list-price': ProductListPrice,
+	'woocommerce/product-list-image': ProductListImage,
+	'woocommerce/product-list-title': ProductListTitle,
+	'woocommerce/product-list-rating': ProductListRating,
+	'woocommerce/product-list-button': ProductListButton,
+	'woocommerce/product-list-summary': ProductListSummary,
+	'woocommerce/product-list-sale-badge': ProductListSaleBadge,
+	...getRegisteredInnerBlocks( blockName ),
+} );
+
+/**
+ * The default layout built from the default template.
+ */
+export const DEFAULT_PRODUCT_LIST_LAYOUT = [
+	{
+		name: 'woocommerce/product-list-sale-badge',
+		props: { align: 'left' },
+	},
+	{
+		name: 'woocommerce/product-list-image',
+		props: {},
+	},
+	{
+		name: 'woocommerce/product-list-title',
+		props: {},
+	},
+	{
+		name: 'woocommerce/product-list-price',
+		props: {},
+	},
+	{
+		name: 'woocommerce/product-list-rating',
+		props: {},
+	},
+	{
+		name: 'woocommerce/product-list-button',
+		props: {},
+	},
+];
+
+/**
+ * The default template (list of inner blocks) for the product list.
+ */
+export const getDefaultBlocks = () => {
+	return DEFAULT_PRODUCT_LIST_LAYOUT.map( ( layout ) => [
+		layout.name,
+		layout.props,
+	] );
+};
```

Comment:
I'm not sure why there is need for `getDefaultBlocks` AND `DEFAULT_PRODUCT_LIST_LAYOUT`.  It appears the only place `getDefaultBlocks` is called is in the `<Editor>` component (`onReset`, `all-products/edit.js`), and it seems you could just loop through `DEFAULT_PRODUCT_LIST_LAYOUT` there directly?

Reply:
Good catch, it could be removed. Should be gone in fddc6feab43254bd1054a124206e78826fcb9d10.